explorer/client/src/components/displaybox/DisplayBox.tsx

```diff
@@ -7,23 +7,41 @@ import {
     FALLBACK_IMAGE,
     ImageModClient,
 } from '../../utils/imageModeratorClient';
-import { transformURL } from '../../utils/stringUtils';
+import { transformURL, extractFileType } from '../../utils/stringUtils';
 
 import styles from './DisplayBox.module.css';
 
-function DisplayBox({ display }: { display: string }) {
+function DisplayBox({
+    display,
+    caption,
+}: {
+    display: string;
+    caption?: string;
+}) {
     const [hasDisplayLoaded, setHasDisplayLoaded] = useState(false);
     const [hasFailedToLoad, setHasFailedToLoad] = useState(false);
 
     const [hasImgBeenChecked, setHasImgBeenChecked] = useState(false);
     const [imgAllowState, setImgAllowState] = useState(false);
 
+    const [hasClickedImage, setHasClickedImage] = useState(false);
+
+    const [fileType, setFileType] = useState('');
+
+    useEffect(() => {
+        extractFileType(display).then((result) => setFileType(result));
```

Comment:
Is there any case where the `display` prop can change before this promise has resolved?  If so, we probably want to guard the set in the promise resolution with an effect cleanup to ensure the set isn't called with old data.